nixos/modules/services/misc/mpdscribble.nix

```diff
@@ -0,0 +1,288 @@
+          default = false;
+          type = types.str;
+          description = "
+            Lastfm Password. Attention: This will be stored in the globally
+            readable /nix/store.
+          ";
+        };
+
+        journal = mkOption {
+          default = "/var/lib/mpdscribble/lastfm.journal";
+          type = types.path;
+          description = ''
+            Journal file for the lastfm backend.
+            Should be changed from the default.
+          '';
+        };
+
+      };
+
+      librefm = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = "
+            Whether to enable the mpdscribble daemon for librefm.
+          ";
+        };
+
+        url  = mkOption {
+          default = false;
+          type = types.str;
+          description = "
+            Whether to enable the mpdscribble daemon for librefm.
+          ";
+        };
+
+        username = mkOption {
+          default = "";
+          type = types.str;
+          description = "
+            Librefm username.
+          ";
+        };
+
+        password = mkOption {
+          default = false;
+          type = types.str;
+          description = "
+            Librefm Password. Attention: This will be stored in the globally
+            readable /nix/store.
+          ";
+        };
+
+        journal = mkOption {
+          default = "/var/lib/mpdscribble/librefm.journal";
+          type = types.path;
+          description = "
+            Journal file for the librefm backend.
+            Should be changed from the default.
+          ";
+        };
+
+      };
+
+      jamendo = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = "
+            Whether to enable the mpdscribble daemon for jamendo.
+          ";
+        };
+
+        url  = mkOption {
+          default = false;
+          type = types.str;
+          description = "
+            Whether to enable the mpdscribble daemon for jamendo.
+          ";
+        };
+
+        username = mkOption {
+          default = "";
+          type = types.str;
+          description = "
+            Jamendo username.
+          ";
+        };
+
+        password = mkOption {
+          default = false;
+          type = types.str;
+          description = "
+            Jamendo Password. Attention: This will be stored in the globally
+            readable /nix/store.
+          ";
+        };
+
+        journal = mkOption {
+          default = "/var/lib/mpdscribble/jamendo.journal";
+          type = types.path;
+          description = "
+            Journal file for the jamendo backend.
+            Should be changed from the default.
+          ";
+        };
+
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+      systemd.services.mpdscribble = {
+        after                   = [ "mpd.service", "network.target" ];
```

Comment:
This will probably just fail

Reply:
Why?

Reply:
List syntax in Nix does not need comma here.

Reply:
ah, sure.